pyproject.toml

```diff
@@ -1,21 +1,48 @@
 [tool.poetry]
 name = "multisignal-epi-inference"
 version = "0.1.0"
-description = "Python package for statistical inference and forecast of epi models using multiple signals."
+description = "The CFA Multisignal Renewal project contains Pyrenew, a Python package for multi-signal Bayesian renewal modeling with NumPyro."
 authors = ["CFA"]
 license = "Apache 2.0"
 readme = "README.md"
-packages = [{include = "multisignal_epi_inference"}]
+packages = [
+    {include = "pyrenew", from = "model/src"},
+]
+include = [{path = "datasets/*.tsv"}]
+exclude = [{path = "datasets/*.rds"}]
+
 
 [tool.poetry.dependencies]
 python = "^3.12"
 numpyro = ">=0.15.1"
+sphinx = "^7.2.6"
+sphinxcontrib-mermaid = "^0.9.2"
+sphinx-autodoc-typehints = "^2.1.0"
+sphinx-book-theme = "^1.1.2"
+jax = ">=0.4.30"
+jaxlib = ">=0.4.30"
+numpy = "^1.26.4"
+polars = "^1.2.1"
+pillow = "^10.3.0"
+nbconvert = "^7.16.4"
+pytest-mpl = "^0.17.0"
+ipywidgets = "^8.1.3"
```

Comment:
I think most of these can be moved to the dev group. I would just keep `jax`, `jaxlib`, `numpy`, `polars`. What is pillow for?

Reply:
My suggestion is

- Basic dependencies are for using the package and running tutorials.
- Dev dependencies are for anything related to precommit (testing, documentation)
- Docs dependencies are for anything specific to the website (I think all the sphinx stuff)

When building the website, it should be possible to install only the sphinx packages without installing numpyro, etc.

Reply:
I am currently working on the `docs` group. On my computer it looks like the below. Before following through with moving of all but `jax`, `jaxlib`, `numpy` and `polars` in the chunk you quoted to `dev`, I would like @damonbayer's perspective, but I tentatively agree. I've forgotten what other package in the above list requires `pillows`.

```toml
[tool.poetry.dependencies]
python = "^3.12"
numpyro = ">=0.15.1"
jax = ">=0.4.30"
jaxlib = ">=0.4.30"
numpy = "^1.26.4"
polars = "^1.2.1"
pillow = "^10.3.0"
nbconvert = "^7.16.4"
pytest-mpl = "^0.17.0"
ipywidgets = "^8.1.3"

[tool.poetry.group.docs]
optional = true

[tool.poetry.group.docs.dependencies]
sphinx = "^7.2.6"
sphinxcontrib-mermaid = "^0.9.2"
sphinx-autodoc-typehints = "^2.1.0"
sphinx-book-theme = "^1.1.2"

[tool.poetry.group.dev]
optional = true

[tool.poetry.group.dev.dependencies]
pyyaml = "^6.0.0"
matplotlib = "^3.8.3"
ipykernel = "^6.29.3"
nbclient = "^0.10.0"
nbformat = "^5.10.0"
pytest-cov = "^5.0.0"
pytest-mpl = "^0.17.0"
numpydoc = "^1.7.0"
arviz = "^0.18.0"
```

Reply:
> When building the website, it should be possible to install only the sphinx packages without installing numpyro, etc.

Maybe I'm missing something, but you do need `numpyro` (and `pyrenew` itself) to build the website if you're going to generate the tutorials. But this is not a problem since that's a basic dependency, and the pattern 

`poetry install --with docs` will also install the basic/default deps.

I agree with the general point!

Reply:
I kind of like having separate `test` and `docs` groups fwiw.

Reply:
Technical comment: I'm fairly certain none of these are actual package deps (versus testing for `pytest-mpl` and docs building for the others)

```
nbconvert = "^7.16.4"
pytest-mpl = "^0.17.0"
ipywidgets = "^8.1.3"
```

Reply:
> Technical comment: I'm fairly certain none of these are actual package deps (versus testing for `pytest-mpl` and docs building for the others)
> 
> ```
> nbconvert = "^7.16.4"
> pytest-mpl = "^0.17.0"
> ipywidgets = "^8.1.3"
> ```

Following George's remarks, I have this uncommitted (need to change the action first, I believe):

```toml
[tool.poetry.dependencies]
python = "^3.12"
numpyro = ">=0.15.1"
jax = ">=0.4.30"
jaxlib = ">=0.4.30"
numpy = "^1.26.4"
polars = "^1.2.1"

[tool.poetry.group.dev]
optional = true

[tool.poetry.group.dev.dependencies]
pyyaml = "^6.0.0"
matplotlib = "^3.8.3"
ipykernel = "^6.29.3"
nbclient = "^0.10.0"
nbformat = "^5.10.0"
pytest-cov = "^5.0.0"
pytest-mpl = "^0.17.0"
numpydoc = "^1.7.0"
arviz = "^0.18.0"
sphinx = "^7.2.6"
sphinxcontrib-mermaid = "^0.9.2"
sphinx-autodoc-typehints = "^2.1.0"
sphinx-book-theme = "^1.1.2"
pillow = "^10.3.0"
nbconvert = "^7.16.4"
ipywidgets = "^8.1.3"
```

Reply:
@AFg6K7h4fhy2 I would encourage you to remove as many of these as possible. For instance, `numpyro` requires `jax`, `jaxlib`, and `numpy`, so I don't think we need to require them here.

Reply:
Mild counterpoint: we use them independently of `numpyro` in the codebase, so I think it's worth having them listed as explicit deps